src/components/application_manager/src/resumption/resumption_data_processor_impl.cc

```diff
@@ -93,22 +93,13 @@ void ResumptionDataProcessorImpl::Restore(
   AddSubscriptions(application, saved_app);
   AddWindows(application, saved_app);
 
-  resumption_status_lock_.AcquireForReading();
   const auto app_id = application->app_id();
-  bool is_requests_list_empty = true;
-  if (resumption_status_.find(app_id) != resumption_status_.end()) {
-    is_requests_list_empty =
-        resumption_status_[app_id].list_of_sent_requests.empty();
-  }
-  resumption_status_lock_.Release();
-
-  if (!is_requests_list_empty) {
+  if (!IsResumptionFinished(app_id)) {
     sync_primitives::AutoWriteLock lock(register_callbacks_lock_);
     register_callbacks_[app_id] = callback;
   } else {
-    SDL_LOG_DEBUG("No requests to HMI for " << app_id
-                                            << " , resumption is successful");
-    callback(mobile_apis::Result::SUCCESS, "Data resumption successful");
+    FinishResumption(callback, app_id);
+    // Todo: create method to erase app data from maps
```

Comment:
@ychernysheva maybe just `EraseAppResumptionData` ?

Reply:
@AKalinich-Luxoft This method needs `function_id` and `corr_id`, but here we need to erase all requests, related to this app. But in fact in EraseAppResumptionData we also erase only last request, and it may be a problem?